allensdk/brain_observatory/ecephys/behavior_ecephys_session.py

```diff
@@ -11,11 +11,149 @@
 
+from allensdk.brain_observatory.behavior.data_files import SyncFile
+from allensdk.brain_observatory.behavior.data_objects.licks import Licks
+from allensdk.brain_observatory.behavior.data_objects import StimulusTimestamps
+from allensdk.brain_observatory.behavior.behavior_session import (
+    StimulusFileLookup)
+from allensdk.brain_observatory.behavior.data_objects.stimuli.stimuli import (
+    Stimuli)
+
+
+class VBNBehaviorSession(BehaviorSession):
+    """
+    A class to create the behavior parts of a VBN session,
+    performing all of the specialized timestamp calculations
+    that implies.
+    """
+
+    @classmethod
+    def from_lims(cls, behavior_session_id: int,
+                  lims_db: Optional[Any] = None,
+                  sync_file: Optional[Any] = None,
+                  monitor_delay: Optional[float] = None,
+                  date_of_acquisition: Optional[Any] = None,
+                  skip_eye_tracking=False,
+                  eye_tracking_z_threshold: float = 3.0,
+                  eye_tracking_dilation_frames: int = 2) \
+            -> "VBNBehaviorSession":
+        raise NotImplementedError(
+                "from_lims is not supported for a VBNBehaviorSession")
+
+    @classmethod
+    def _read_stimuli(
+            cls,
+            stimulus_file_lookup: StimulusFileLookup,
+            stimulus_timestamps: StimulusTimestamps,
+            stimulus_presentation_columns: Optional[List[str]] = None
+    ) -> Stimuli:
+        raise NotImplementedError(
+            "VBNBehaviorSessions read their stimulus tables from "
+            "a precomputed csv file; they should not be computed "
+            "on the fly by the AllenSDK")
+
+    @classmethod
+    def _read_behavior_stimulus_timestamps(
+            cls,
+            stimulus_file_lookup: StimulusFileLookup,
+            sync_file: Optional[SyncFile],
+            monitor_delay: float) -> StimulusTimestamps:
+        """
+        Assemble the StimulusTimestamps from the SyncFile.
+        If a SyncFile is not available, use the
+        behavior_stimulus_file
+        """
+        timestamps = StimulusTimestamps.from_multiple_stimulus_blocks(
+                sync_file=sync_file,
+                list_of_stims=[
+                     stimulus_file_lookup.behavior_stimulus_file,
+                     stimulus_file_lookup.mapping_stimulus_file,
+                     stimulus_file_lookup.replay_stimulus_file],
+                stims_of_interest=[0, ],
+                monitor_delay=monitor_delay)
+        return timestamps
+
+    @classmethod
+    def _read_session_timestamps(
+            cls,
+            stimulus_file_lookup: StimulusFileLookup,
+            sync_file: Optional[SyncFile],
+            monitor_delay: float) -> StimulusTimestamps:
+        """
+        Assemble the StimulusTimestamps (with monitor delay) that will
+        be associated with this session
+        """
+        timestamps = StimulusTimestamps.from_multiple_stimulus_blocks(
```

Comment:
Why not consolidate `_read_session_timestamps` and `_read_behavior_stimulus_timestamps` by accepting an argument `stims_of_interest` and having one of these methods call the other by passing in a different value for `stims_of_interest`? It makes it clearer how these 2 methods are different.